Tests/TwoDimension/AreaTests.cs

```diff
@@ -170,15 +170,66 @@ public void PositionSet()
 		{
-			// TODO: Issue 6 (https://github.com/Wizcorp/TileSystem/issues/6)
-			Assert.Fail();
-		}
+            Area area = new Area();
+            var mockLevel = new Mock<ILevel>();
+            var mockPosition = new Mock<IPosition2D>();
+            var mockTile = new Mock<Tile>();
+            var tilePosition = new Position2D(0, 0);
+            var tilePositionNotFound = new Position2D(1, 0);
+
+            area.SetPosition(mockLevel.Object, mockPosition.Object);
+            mockTile.Object.SetPosition(area, tilePosition);
+            area.Add(mockTile.Object);
+
+            // Test Nulls
+            Assert.That(() => area.Get(null), Throws.ArgumentNullException);
+
+            // Test Same Tile
+            Assert.AreSame(mockTile.Object, area.Get(tilePosition));
+
+            // Test Tile Not Found
+            Assert.AreNotSame(mockTile.Object, area.Get(tilePositionNotFound));
+        }
 
 		[Test]
 		public void PositionGetNeighbours()
 		{
-			// TODO: Issue 6 (https://github.com/Wizcorp/TileSystem/issues/6)
-			Assert.Fail();
-		}
-	}
+            Area area = new Area();
+            var mockLevel = new Mock<ILevel>();
+            var mockPosition = new Mock<IPosition2D>();
+            var mockTile = new Mock<Tile>();
+            var tilePosition = new Position2D(0, 0);
+            var mockTileNotInArea = new Mock<Tile>();
+            var mockFirstNeighbourTile = new Mock<Tile>();
+            var firstNeighbourTilePosition = new Position2D(1, 0);
+            var mockSecondNeighbourTile = new Mock<Tile>();
+            var secondNeighbourTilePosition = new Position2D(1, 1);
+            var mockNotANeighbourTile = new Mock<Tile>();
+            var notANeighbourTilePosition = new Position2D(4, 0);
+
+            area.SetPosition(mockLevel.Object, mockPosition.Object);
+            mockTile.Object.SetPosition(area, tilePosition);
+            area.Add(mockTile.Object);
+
+            // Test Nulls
+            Assert.That(() => area.GetNeighbours(null), Throws.ArgumentNullException);
+
+            // Test Tile Not In Area
+            Assert.That(() => area.GetNeighbours(mockTileNotInArea.Object), Throws.ArgumentException);
+
+            // Test No Neighbours
+            Assert.Null(area.GetNeighbours(mockTile.Object));
+
+            // Test One Or More Neighbours
+            mockFirstNeighbourTile.Object.SetPosition(area, firstNeighbourTilePosition);
+            area.Add(mockFirstNeighbourTile.Object);
+            Assert.IsTrue(area.GetNeighbours(mockTile.Object).Count == 1);
+            mockSecondNeighbourTile.Object.SetPosition(area, secondNeighbourTilePosition);
+            area.Add(mockSecondNeighbourTile.Object);
+            Assert.IsTrue(area.GetNeighbours(mockTile.Object).Count == 2);
+            mockNotANeighbourTile.Object.SetPosition(area, notANeighbourTilePosition);
+            area.Add(mockNotANeighbourTile.Object);
+            Assert.IsTrue(area.GetNeighbours(mockTile.Object).Count == 2);
```

Comment:
Do you think this is an adequate test? For instance is someone is making this change and it happens that the new change actually returns two wrong tiles rather than the two tiles you are expecting.

I'd like to hear your opinions on this because Unit Tests are quite a hot potato topic in the company, it's nice to hear other peoples views on how verbose Unit Tests should be.

Reply:
This test would be irrelevant if someone were to change the test, or if the GetNeighbours method would change to return only the direct neighbours (left, top, right, bottom).

One way to fix this would be to check if the tiles we know are neighbours to the tested tile are present in the list returned by the method.

Reply:
`One way to fix this would be to check if the tiles we know are neighbours to the tested tile are present in the list returned by the method.`

This is exactly what I was getting at. Do you believe this is better or worse in this case?

Reply:
Since the method may change, I believe the actual test is not reliable and should be rewriten.